src/game/mario_actions_automatic.c

```diff
@@ -725,7 +725,11 @@ s32 act_in_cannon(struct MarioState *m) {
             marioObj->oMarioCannonInputYaw -= (s16)(m->controller->stickX * 10.0f);
 
             m->faceAngle[0] = CLAMP(m->faceAngle[0], 0, DEGREES(80));
+    #ifdef UNRESTRICTED_CANNON_AIMING
+            marioObj->oMarioCannonInputYaw = marioObj->oMarioCannonInputYaw;
+    #else
```

Comment:
```suggestion
    #ifndef UNRESTRICTED_CANNON_AIMING
```
Why not just this?

Reply:
either this makes no sense or I'm not understanding

you need the code in the ifdef for unrestricted cannon aiming and the code in the else for restricted cannon aiming, this wouldn't work

Reply:
It's just setting `marioObj->oMarioCannonInputYaw` to itself.

Reply:
500 iq code

(might also be the case that that's a typo and it's actually supposed to set something else)

Reply:
OH MY GOD